src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb

```diff
@@ -173,7 +173,9 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
                 Dim iBaseAddress As UInteger
 
-                If (TypeOf (value) Is UInteger) Then
+                Dim throwAwayObject As Decimal = Nothing
+                If (TypeOf (value) Is UInteger OrElse
+                    (CpsPropertyDescriptorWrapper.IsAnyCpsComponent(m_Objects) AndAlso Decimal.TryParse(DirectCast(value, String), throwAwayObject))) Then
                     iBaseAddress = CUInt(value)
```

Comment:
@jviau BaseAddress property is currently backed by a StringProperty in the XAML. I still get a string value, if I change the Property type to IntProperty from the StringProperty. For eg: `"123456"`

As it turns out, to obtain the value, DynamicType's [GetValue](http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&leftSymbol=qmr1zelnwqn9&file=PropertyPages%5CDynamicType.cs) is called, which always return a string. Is there a way it could return the actual type that the property is initially backed with?